analyses/infercnv-consensus-cell-type/template-notebooks/SCPCP000015_explore-infercnv-results.Rmd

```diff
@@ -0,0 +1,418 @@
+```
+
+
+## UMAPs
+
+Next, we'll visualize the total CNV counts as a UMAP.
+Below, the UMAP colored by cell type is shown on the left, and the UMAP colored by total CNV counts is shown on the right.
+Ideally, unknown and tumor cells will overlap with regions in the UMAP with higher total CNV.
+
+
+```{r, fig.height = 10}
+celltype_umap <- ggplot(umap_df) +
+  aes(x = UMAP1, y = UMAP2, color = celltype) +
+  geom_point(size = 0.1, alpha = 0.3) +
+  guides(color = guide_legend(override.aes = list(size = 2, alpha = 1))) +
+  labs(title = "Cell types") +
+  umap_theme
+
+cnv_umap <- ggplot(query_has_cnv_df) +
+  aes(x = UMAP1, y = UMAP2, color = total_cnv_per_cell) +
+  geom_point(size = 0.1, alpha = 0.3) +
+  scale_color_viridis_c() +
+  labs(title = "Total CNV per cell") +
+  umap_theme
+
+celltype_umap / cnv_umap
+```
+
+
+## Clusters
+
+Part of `inferCNV`'s algorithm involves clustering the data, and poor clusters have the potential to lead to inaccurate inferences of CNV.
+To see whether we'd like to further tune the clustering parameters, we'll get a sense of their size and rough composition here.
+
+How many clusters are there, and how big are they?
+We'll show this separately for reference vs query cells.
+
+```{r}
+cluster_count_df <- has_cnv_df |>
+  dplyr::count(subcluster, cell_group)
+ggplot(cluster_count_df) +
+  aes(x = n, fill = cell_group) +
+  geom_histogram(color = "grey20", bins = 15) +
+  facet_wrap(vars(cell_group), scales = "free")
+```
+
+Is there any relationship between cluster size and the number of CNVs?
+We would ideally not want there to be a relationship since this might indicate a bias.
+
+```{r}
+cluster_size_df <- has_cnv_df |>
+  dplyr::add_count(subcluster, name = "cluster_size") |>
+  dplyr::select(
+    total_cnv_per_cell,
+    cluster_size
+  ) |>
+  dplyr::distinct()
+
+ggplot(cluster_size_df) +
+  aes(x = cluster_size, y = total_cnv_per_cell) +
+  geom_point() +
+  geom_smooth(method = "lm")
+```
+
+
+Let's also look more specifically at the `reference` clusters, where we might expect some batch effects since cells were taken from several different libraries.
+Do originating libraries cluster together?
+When interpreting this plot, bear in mind that there might also be an association between cell types and libraries, and we would expect cell types to still cluster together.
+
+```{r}
+reference_cells_df <- has_cnv_df |>
+  dplyr::filter(cell_group == "reference")
+
+ggplot(reference_cells_df) +
+  aes(x = subcluster, fill = library_id) +
+  geom_bar() +
+  theme(
+    legend.position = "bottom",
+    axis.text.x = element_blank()
+  )
+```
+
+
+## CNV across chromosomes
+
+Now, we'll look at CNVs across chromosomes, specifically considering the `proportion_scaled_cnv_` output from the HMM.
+
+Again, these are the specific CNVs we expect to see in Ewing sarcoma:
+
+```{r}
+validation_df
+```
+
+```{r}
+# First, construct data frame with proportion_scaled_ instead of has_cnv
```

Comment:
I think it's fine to leave this plot in, but it is kind of busy. 

What I would like to see is just the total CNV proportion for each chromosome with lines colored by cell type annotation. Again, I would use those three groups of reference, tumor cells, and non-tumor and non-reference.